src/websocket/base.ts

```diff
@@ -146,6 +145,29 @@ abstract class BaseWebSocket extends EventEmitter {
     this.ws.onmessage = (evt) => this.onMessage(evt);
     this.ws.onerror = (evt) => this.onError(evt);
     this.ws.onclose = () => this.onClose();
+
+    this.started = true;
+  }
+
+  /**
+   * Sets all event listeners to noops on the WebSocket instance 
+   * and close it.
+   **/
+  closeWs() {
```

Comment:
Should we add a test that calls the methods just like the race condition and checks that it behaved correctly?

Reply:
Yes we should, added in a80eb84098e7c4aea5117aa6770cdbc0415c9d89

Reply:
Great! Thanks! Does this test fail without the fixes in this PR?

Reply:
Yes it does

Reply:
Thanks!